### PR TITLE
fix(reg-exp-router): use matchers[METHOD_NAME_ALL] as fallback for unknown method

### LINKED ISSUE #2254: Exception occured when send invalid HTTP method to RegExpRouter
### What version of Hono are you using?

4.0.5

### What runtime/platform is your app running on?

deno

### 1. Run HTTP Server

Run below script by `deno run --allow-net`

```ts
import { Hono, RegExpRouter, SmartRouter } from "https://deno.land/x/hono@v4.0.5/mod.ts";

const app = new Hono({
    router: new SmartRouter(
        {routers: [new RegExpRouter()]}
    )
});

app.all("/", c => c.text("Hello Hono!"));

Deno.serve(app.fetch);
```

### 2. Send request with invalid HTTP method

```sh
curl -X "FOOBAR" "http://localhost:8000"
```

Hono returns `500 Internal Server Error` and dumps error log below.

```
TypeError: Cannot read properties of undefined (reading '2')
    at RegExpRouter.match (https://deno.land/x/hono@v4.0.5/router/reg-exp-router/router.ts:217:34)
    at RegExpRouter.match (https://deno.land/x/hono@v4.0.5/router/reg-exp-router/router.ts:231:17)
    at SmartRouter.match (https://deno.land/x/hono@v4.0.5/router/smart-router/router.ts:37:22)
    at Hono.matchRoute (https://deno.land/x/hono@v4.0.5/hono-base.ts:300:24)
    at Hono.dispatch (https://deno.land/x/hono@v4.0.5/hono-base.ts:323:30)
    at fetch (https://deno.land/x/hono@v4.0.5/hono-base.ts:375:17)
    at ext:deno_http/00_serve.js:455:24
    at ext:deno_http/00_serve.js:689:29
    at eventLoopTick (ext:core/01_core.js:65:7)
```

### 3. send valid request

```sh
curl "http://localhost:8000"
```

It is valid request, But Hono returns `500 Internal Server Error` and dumps error.

```
Error: Can not add a route since the matcher is already built.
    at RegExpRouter.add (https://deno.land/x/hono@v4.0.5/router/reg-exp-router/router.ts:137:13)
    at https://deno.land/x/hono@v4.0.5/router/smart-router/router.ts:35:18
    at Array.forEach (<anonymous>)
    at SmartRouter.match (https://deno.land/x/hono@v4.0.5/router/smart-router/router.ts:34:16)
    at Hono.matchRoute (https://deno.land/x/hono@v4.0.5/hono-base.ts:300:24)
    at Hono.dispatch (https://deno.land/x/hono@v4.0.5/hono-base.ts:323:30)
    at fetch (https://deno.land/x/hono@v4.0.5/hono-base.ts:375:17)
    at ext:deno_http/00_serve.js:455:24
    at ext:deno_http/00_serve.js:689:29
    at eventLoopTick (ext:core/01_core.js:65:7)
```

### What is the expected behavior?

_No response_

### What do you see instead?

_No response_

### Additional information

The error of step 3. only occured when using `SmartRouter` with `RegExpRouter`.

### PR DESCRIPTION
**Pull request by usualoma**
fixes #2254

Fixed an error when accessed by an unknown method. I don't think there is a compatibility issue.

I will think about #1953 later.

### Author should do the followings, if applicable

- [x] Add tests
- [x] Run tests
- [x] `yarn denoify` to generate files for Deno

### CODE REVIEW

File: src/router/reg-exp-router/router.test.ts [hunk 1]
```diff
@@ -678,3 +678,13 @@ describe('Capture Group', () => {
     })
   })
 })
+
+describe('Unknown method', () => {
+  const router = new RegExpRouter<string>()
+  router.add('GET', '/', 'index')
```
Comment (yusukebe) [inline src/router/reg-exp-router/router.test.ts:684]: Should we have the `ALL` pattern?

```ts
router.add('ALL', '/all', 'index')

// ...

const [res] = router.match('HONO', '/all')
// ...
```

**Comment by yusukebe** [comment]
@usualoma 

Thanks! I've let one comment.

**Comment by usualoma** [comment]
@yusukebe Thanks! I have added 802d176, how about this?

**Comment by usualoma** [comment]
+ b9eb1cc

**Review by yusukebe — APPROVED** [review]
Looks good!

**Comment by yusukebe** [comment]
@usualoma 

Thanks. Merge now!

**Pull request merged**